components/instructions/ExecuteAllInstructionButton.tsx

```diff
@@ -0,0 +1,147 @@
+import { useEffect, useState } from 'react'
+import {
+  InstructionExecutionStatus,
+  Proposal,
+  ProposalTransaction,
+  ProposalState,
+} from '@solana/spl-governance'
+import React from 'react'
+import { CheckCircleIcon, PlayIcon, RefreshIcon } from '@heroicons/react/solid'
+import Button from '@components/Button'
+import { RpcContext } from '@solana/spl-governance'
+import useRealm from '@hooks/useRealm'
+import useWalletStore from 'stores/useWalletStore'
+import { ProgramAccount } from '@solana/spl-governance'
+import { PublicKey } from '@solana/web3.js'
+import Tooltip from '@components/Tooltip'
+import { getProgramVersionForRealm } from '@models/registry/api'
+import { executeInstructions } from 'actions/executeInstructions'
+
+export enum PlayState {
+  Played,
+  Unplayed,
+  Playing,
+  Error,
+}
+
+export function ExecuteAllInstructionButton({
+  proposal,
+  playing,
+  setPlaying,
+  proposalInstructions,
+}: {
+  proposal: ProgramAccount<Proposal>
+  proposalInstructions: ProgramAccount<ProposalTransaction>[]
+  playing: PlayState
+  setPlaying: React.Dispatch<React.SetStateAction<PlayState>>
+}) {
+  const { realmInfo } = useRealm()
+  const wallet = useWalletStore((s) => s.current)
+  const connection = useWalletStore((s) => s.connection)
+  const fetchRealm = useWalletStore((s) => s.actions.fetchRealm)
+  const connected = useWalletStore((s) => s.connected)
+
+  const [currentSlot, setCurrentSlot] = useState(0)
+
+  const canExecuteAt = proposal?.account.votingCompletedAt
+    ? proposal.account.votingCompletedAt.toNumber() + 1
+    : 0
+
+  const ineligibleToSee = currentSlot - canExecuteAt >= 0
+
+  const rpcContext = new RpcContext(
+    new PublicKey(proposal.owner.toString()),
+    getProgramVersionForRealm(realmInfo!),
+    wallet!,
+    connection.current,
+    connection.endpoint
+  )
+
+  useEffect(() => {
+    if (ineligibleToSee && proposal) {
+      const timer = setTimeout(() => {
+        rpcContext.connection.getSlot().then(setCurrentSlot)
+      }, 5000)
```

Comment:
What does this effect do? Why is it required? Can we remove it? Can it be replaced by something better? Why 5000ms? Add a comment in any case

Reply:
No idea. Copy pasted the code of `components/instructions/ExecuteInstructionButton.tsx` and replaced single Instruction by Instruction array

Reply:
Then we should look into the git history maybe there's a hint or a PR review that explains why it was added. If no reason found, if it works without it then I guess it's best, if not, just say that it was copy pasta and it doesn't work without

Reply:
it refreshes the current slot every 5sec in the rpcContext. It prevents from timing out when building and confirm the tx I think

Reply:
yes ser